pandas/core/dtypes/dtypes.py

```diff
@@ -560,17 +581,18 @@ def categories(self):
         """
         An ``Index`` containing the unique categories allowed.
         """
-        return self._categories
+        from pandas import Index
+        return typing.cast(Index, self._categories)
 
     @property
-    def ordered(self):
+    def ordered(self) -> Optional[bool]:
```

Comment:
Is this optional? I thought ordered was always sure or values (categories are optional).

Reply:
Yes:

```python
In [1]: pd.CategoricalDtype()
Out[1]: CategoricalDtype(categories=None, ordered=None)
```
I think this has somwthing to to with updating dtypes (don't update None, but do update False).

Reply:
grr, this isn't needed, and isn't how we document things. https://github.com/pandas-dev/pandas/issues/26336